pkg/engine/plugins.go

```diff
@@ -43,6 +43,80 @@ const (
 	preparePluginVerboseLog = 8
 )
 
+// PackageSet represents a set of packages.
+type PackageSet map[string]workspace.PackageDescriptor
+
+// Add adds a package to this package set.
+func (p PackageSet) Add(plug workspace.PackageDescriptor) {
+	p[plug.String()] = plug
+}
+
+// NewPackageSet creates a new empty pluginSet.
+func NewPackageSet(plugins ...workspace.PackageDescriptor) PackageSet {
+	var s PackageSet = make(map[string]workspace.PackageDescriptor, len(plugins))
+	for _, p := range plugins {
+		s.Add(p)
+	}
+	return s
+}
+
+// Union returns the union of this PackageSet with another PackageSet.
+func (p PackageSet) Union(other PackageSet) PackageSet {
+	newSet := NewPackageSet()
+	for _, value := range p {
+		newSet.Add(value)
+	}
+	for _, value := range other {
+		newSet.Add(value)
+	}
+	return newSet
+}
```

Comment:
Does this mean we can delete `PluginSet.Union`?

Reply:
Yes